src/compiler/compiler.ts

```diff
@@ -59,12 +59,15 @@ export async function compileAll(
     tsCompile(files, opts?.tsconfig, opts?.dts);
   }
 
-  // Remove .ts entries if ts option was not set
-  if (!opts?.ts) {
-    for (const [fileName] of files) {
-      if (fileName.endsWith(".ts")) {
-        files.delete(fileName);
-      }
+  // Remove corresponding entries if format option was not set
+
+  for (const [fileName] of files) {
+    if (
```

Comment:
```suggestion
    for (const [fileName] of files) {
      if (fileName.endsWith(".ts") && !fileName.endsWith(".ts")) {
        files.delete(fileName);
      }
```

if you choose not to generate the .js above